hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/LongVarcharJdbcType.java

```diff
@@ -39,6 +46,11 @@ public int getJdbcTypeCode() {
 		return jdbcTypeCode;
 	}
 
+	@Override
+	public int getDefaultSqlTypeCode() {
+		return defaultSqlTypeCode;
+	}
```

Comment:
Now that I understand, I would say we don't need to parameterize the class by `defaultSqlTypeCode`, we can just hardcode this method to return `Types.LONGVARCHAR`, right?

Reply:
I guess that this is, at least, partially correct. Only it seems that JDBC type code should be fixed, and default SQL code changeable, but I will check that again.

Reply:
The one that is used for binding to `null` values to JDBC is that one that needs to be hardcoded. The one used for registration and for schema export is the one that's parameterized.

Reply:
I guess now that is exactly this